src/llm/agent-evals/slack-agent/index.spec.ts

```diff
@@ -0,0 +1,139 @@
+import { describe, it, beforeAll, afterAll } from '@jest/globals';
+import { createTrajectoryMatchEvaluator } from 'agentevals';
+import { QuixAgent } from '../../quix-agent';
+import { createSlackToolsExport } from '@clearfeed-ai/quix-slack-agent';
+import { createSlackMockedTools, ToolResponseTypeMap } from './mock';
+import type { AvailableToolsWithConfig, LLMContext } from '@quix/llm/types';
+import { Logger } from '@nestjs/common';
+import { testCases } from './test-data';
+import * as fs from 'fs';
+import * as path from 'path';
+import { TestRunDetail } from '../common/types';
+import { AIMessage } from '@langchain/core/messages';
+import { getTestOpenAIProvider } from '../common/utils';
+import { isEmpty, isNumber, isString } from 'lodash';
+
+describe('QuixAgent Slack – real LLM + mocked tools', () => {
+  let agent: QuixAgent;
+  let slackToolsDef: ReturnType<typeof createSlackToolsExport>;
+  let llm: ReturnType<typeof getTestOpenAIProvider>;
+  let evaluator: ReturnType<typeof createTrajectoryMatchEvaluator>;
+  const allTestRunDetails: TestRunDetail[] = [];
+  const slackConfig = {
+    token: 'dummy-token',
+    teamId: 'T123'
+  };
+
+  beforeAll(() => {
+    slackToolsDef = createSlackToolsExport(slackConfig);
+
+    llm = getTestOpenAIProvider(process.env.OPENAI_API_KEY);
+
+    evaluator = createTrajectoryMatchEvaluator({
+      trajectoryMatchMode: 'superset',
+      toolArgsMatchMode: 'superset',
+      toolArgsMatchOverrides: {
+        slack_post_message: (a, b) => {
+          // a - Actual - AI generated
+          // b - Expected
+          return isString(a.text) && !isEmpty(a.text) && a.channel_id === b.channel_id;
+        },
+        slack_get_channel_history: (a, b) => {
+          return isNumber(a.limit) && a.limit > 0 && a.channel_id === b.channel_id;
+        }
+      }
```

Comment:
Please use better variable names
```suggestion
            toolArgsMatchOverrides: {
        slack_post_message: (actualToolCalArguments, referenceToolCallArguments) => {
          return (
            isString(actualToolCalArguments.text) &&
            !isEmpty(actualToolCalArguments.text) &&
            actualToolCalArguments.channel_id === referenceToolCallArguments.channel_id
          );
        },
        slack_get_channel_history: (actualToolCalArguments, referenceToolCallArguments) => {
          return (
            isNumber(actualToolCalArguments.limit) &&
            actualToolCalArguments.limit > 0 &&
            actualToolCalArguments.channel_id === referenceToolCallArguments.channel_id
          );
        }
      }
```